--- riot/device_list.sh.orig	2017-09-30 02:06:28.509368539 +0200
+++ riot/device_list.sh	2017-09-30 02:35:03.679822014 +0200
@@ -89,7 +89,7 @@
             if [ $num_mgmt_sriov -gt 0 ]; then
                 pci_list=$(lspci | grep "$sriov_dev_str" | sed "1,${num_mgmt_sriov}d" | awk '{ print $1 }')
             else
-                pci_list=$(lspci | grep "$sriov_dev_str" | awk '{ print $1 }')
+                pci_list=$PCI $(lspci | grep "$sriov_dev_str" | awk '{ print $1 }')
             fi
             # check for additional virtio interfaces
             if [ $vmx_junos_device_type_virtio -gt ${num_mgmt_virtio} ];
@@ -134,8 +134,8 @@
             numports=$vmx_junos_device_type_vmxnet3
             device_type=$DEVICE_TYPE_VMXNET3
         elif [ ${vmx_junos_device_count_xenpv} -gt 0 ]; then
-            trace "Detected Junos Device type vif/veth"
-            numports=$((${numports} + ${vmx_junos_device_count_xenpv}))
+            trace "Detected Junos Device type vif/veth/sriov"
+            numports=$((${numports} + ${vmx_junos_device_count_xenpv} + $(echo "$PCI" | wc -w)))
             vdev_str=${vmx_junos_device_list_xenpv}
             device_type=$DEVICE_TYPE_VIF
         fi 
