--- riot/start_riot.sh.orig	2018-03-17 00:44:22.000000000 +0000
+++ riot/start_riot.sh	2018-07-23 06:38:19.198246425 +0000
@@ -126,7 +126,7 @@
     ip link set dev veth-vmxt up
 fi
 
-pci_whitelist=""
+pci_whitelist=$(for id in $PCI; do echo -w $id ;done)
 if [ -n "$pci_list" ]; then
     pci_whitelist=$(for id in $pci_list; do echo -w $id ;done)
 fi
@@ -155,7 +155,7 @@
     fi
 else
     # How many cores do we have?
-    numcpus=`lscpu | grep "^CPU(s):" | awk '{ print $NF }'`
+    numcpus=`nproc`
     case $numcpus in
     2 ) 
       # Fake the args for 3 cpus, but pin io and worker to core 1
@@ -167,6 +167,23 @@
       eal_args="--lcores='(0-2)@0'"
       numcpus=3
     ;;
+    * )
+      # map logical to physical cores
+      function mapcores {
+        i=0
+        for range in $cores; do
+          if [ -z "${range##*-*}" ] ;then
+            range=$(echo $range | sed "s/-/ /")
+            for j in $(seq $range); do echo -n ",$i@$j"; i=$(($i + 1)); done
+          else
+            echo -n ",$i@$range"; i=$(($i + 1))
+          fi
+        done
+      }
+      cores=$(cat /proc/self/status |grep Cpus_allowed_list| awk '{print$2}' | sed "s/,/ /g")
+      list=$(mapcores | sed "s/,//")
+      eal_args="--lcores='$list'"
+
     esac
 
     arg_str=${arg_str}`/home/pfe/riot/get_virtio_argstr $numcpus $numports $lo_num $fab_enable $pfe_id $vmx_phase $service_pic_enable $src_mac $dst_mac`
@@ -200,7 +217,7 @@
 
 # Get the taskset mask
 
-taskset_mask=`lscpu -x | grep "On-line CPU(s) mask:" | awk '{ print $NF }'`
+taskset_mask="0x`cat /proc/self/status |grep Cpus_allowed:|awk {'print $2'}|tr -d ,`"
 
 # Reference command string.
 #python riot.py -c fff0 -n 2 -- --rx "(0,0,0,8),(1,0,2,9)" --tx "(0,8),(1,9)" --w "10,11,12,13" --rpio "inet,3000,3001" --hostif "inet,3002" --bsz "(32,32),(32,32),(64,64)" 
@@ -219,7 +236,11 @@
        riot_args=$riot_args_from_conf
     fi
 fi
-
+pci_whitelist="--no-pci -m 1024"
 CMDLINE="$CMD -c $taskset_mask -n 2 $eal_args --log-level=$debug_level "$vdev_str" $pci_whitelist -- $arg_str $vlan_arg --rpio \"$proto,3000,3001\" --hostif \"$proto,3002\" --bsz \"(32,32),(32,32),(32,32)\" $riot_args"
 trace "$CMDLINE"
-eval $CMDLINE
+while true; do
+  echo $CMDLINE && eval $CMDLINE
+  echo "restarting riot in 5 seconds ..."
+  sleep 5
+done
