--- riot/dpdk/dpdk_setup.sh.orig
+++ riot/dpdk/dpdk_setup.sh
@@ -17,15 +17,6 @@ create_mnt_huge()
         fi
 }

-if [ "${docker}" == "yes" ]
-then
-    echo "Setting Up DPDK for Docker env"
-    # Hugepages should be setup on the host
-    #  We only set the mount point here
-    create_mnt_huge
-    # nothing else to be done for Docker
-    exit
-fi

 trace "SETTING UP DPDK ENVIRONMENT ON"
 uname -a
